### PR TITLE
Move tracking issue due dates to Issues Someone Else Cares About project

### PR DESCRIPTION
**Pull request by hubertdeng123**
a duplicate of https://github.com/getsentry/eng-pipes/pull/481

Need to time this merge better, after slack notifications are updated to use project field

### CODE REVIEW

File: src/brain/issueLabelHandler/followups.ts [hunk 1]
```diff
@@ -158,5 +169,25 @@ export async function ensureOneWaitingForLabel({
...
+  let timeToRespondBy;
+  if (labelName === WAITING_FOR_PRODUCT_OWNER_LABEL) {
+    timeToRespondBy =
+      (await calculateSLOViolationTriage(UNTRIAGED_LABEL, issue.labels)) ||
+      moment().toISOString();
+  } else if (labelName === WAITING_FOR_SUPPORT_LABEL) {
+    timeToRespondBy =
+      (await calculateSLOViolationRoute(UNROUTED_LABEL)) ||
+      moment().toISOString();
+  } else {
```
Comment (chadwhitacre) [inline src/brain/issueLabelHandler/followups.ts:181]: Should we care about these missing tests?

File: src/brain/issueLabelHandler/route.ts [hunk 2]
```diff
@@ -101,64 +91,33 @@ export async function markUnrouted({
...
     await octokit.teams.getByName({
       org: SENTRY_ORG,
       team_slug: ghTeamSlug,
     }); // expected to throw if team doesn't exist
-    return `Routing to @${SENTRY_ORG}/${ghTeamSlug} for [triage](https://develop.sentry.dev/processing-tickets/#3-triage)`;
+    return `Routing to @${SENTRY_ORG}/${ghTeamSlug} for [triage](https://develop.sentry.dev/processing-tickets/#3-triage) ⏲️`;
   } catch (error) {
     Sentry.captureException(error);
-    return `Failed to route for ${productAreaLabelName}. Defaulting to @${SENTRY_ORG}/open-source for [triage](https://develop.sentry.dev/processing-tickets/#3-triage)`;
+    return `Failed to route for ${productAreaLabelName}. Defaulting to @${SENTRY_ORG}/open-source for [triage](https://develop.sentry.dev/processing-tickets/#3-triage) ⏲️`;
```
Comment (chadwhitacre) [inline src/brain/issueLabelHandler/route.ts:117]: I guess we'll update our "route" and "triage" nomenclature later.

File: src/brain/issueLabelHandler/route.ts [hunk 3]
```diff
@@ -181,7 +140,7 @@ export async function markRouted({
   const owner = payload.repository.owner.login;
   const octokit = await getClient(ClientType.App, owner);
   const labelsToRemove: string[] = [];
-  const labelNames = issue?.labels?.map(label => label.name) || [];
+  const labelNames = issue?.labels?.map((label) => label.name) || [];
```
Comment (chadwhitacre) [inline src/brain/issueLabelHandler/route.ts:143]: ?

Comment (hubertdeng123) [reply src/brain/issueLabelHandler/route.ts:143]: doesn't do anything, I can definitely change this back

File: src/brain/issueLabelHandler/followups.ts (hunk 1 above)
Comment (hubertdeng123) [reply src/brain/issueLabelHandler/followups.ts:181]: wrote some tests to cover those edge cases

File: src/brain/issueLabelHandler/route.ts [hunk 4]
```diff
@@ -67,6 +67,36 @@ function shouldLabelBeRemoved(labelName, target_name) {
   );
 }
 
+async function getReadableTimeStamp(timeToTriageBy, productAreaLabelName) {
```
Comment (hubertdeng123) [inline src/brain/issueLabelHandler/route.ts:70]: thought this would be a better place to put this, since it's above all the functions that listen to webhooks

File: src/utils/githubEventHelpers.ts [hunk 5]
```diff
@@ -105,7 +105,7 @@ export async function modifyProjectIssueField(
     octokit
   );
   const singleSelectOptionId = projectFieldNodeIDMapping[projectFieldOption];
-  const addIssueToGlobalIssuesProjectMutation = `mutation {
+  const modifyProjectIssueFieldMutation = `mutation {
```
Comment (chadwhitacre) [inline src/utils/githubEventHelpers.ts:108]: Fixing a bad name?

Comment (hubertdeng123) [reply src/utils/githubEventHelpers.ts:108]: yessir, this function definitely has nothing to do with adding issues to global issues project

File: src/brain/issueLabelHandler/route.ts [hunk 6]
```diff
@@ -131,36 +162,6 @@ async function routeIssue(octokit, productAreaLabelName) {
...
-      ? 'dddd, MMMM Do [at] HH:mm'
-      : 'dddd, MMMM Do [at] h:mm a';
-  return {
-    readableDueByDate: dueByMoment
-      .tz(OFFICE_TIME_ZONES[lastOfficeInBusinessHours])
-      .format(officeDateFormat),
-    lastOfficeInBusinessHours,
-  };
-}
-
```
Comment (chadwhitacre) [inline src/brain/issueLabelHandler/route.ts:163]: Are there material changes to this function? It looks like it's just moving to a different place in the file.

Comment (hubertdeng123) [reply src/brain/issueLabelHandler/route.ts:163]: see https://github.com/getsentry/eng-pipes/pull/488#discussion_r1237446925 😄

File: src/brain/issueLabelHandler/followups.ts [hunk 7]
```diff
@@ -158,5 +169,25 @@ export async function ensureOneWaitingForLabel({
...
+    timeToRespondBy = '';
+  }
```
Comment (chadwhitacre) [inline src/brain/issueLabelHandler/followups.ts:183]: Probably not the biggest deal, but aren't we running `calculateSLOViolation*` twice, once for the comment and once for the project field? Won't they be slightly different values when conceptually they should be the same?

Comment (hubertdeng123) [reply src/brain/issueLabelHandler/followups.ts:183]: Yes, for now they may differ by a factor of milliseconds but for our purposes that shouldn't be a big deal since to our logic and we will remove the timestamp from comments soon. I don't see a way where we can fold that logic into one right now since they exist in two different webhook listeners

File: src/brain/issueLabelHandler/route.ts (hunk 6 above)
Comment (chadwhitacre) [reply src/brain/issueLabelHandler/route.ts:163]: 😁

**Review by chadwhitacre — APPROVED** [review]
